CHANGELOG.md

```diff
@@ -57,8 +57,19 @@ for Rust libraries in [RFC #1105](https://github.com/rust-lang/rfcs/blob/master/
   use associated constants where appropriate.
 
 * You will now need to invoke `allow_tables_to_appear_in_same_query!` any time two tables
-  appear together in the same query, even if there is a `joinable!` invocation
-  for those tables.
+  appear together in the same query, even if there is a `joinable!` invocationfor those tables.
+
+* The story about the custom derive annotation changed. All pure custom
+ derive procedural macro implementations (`Queryable`, `QueryableByName`,
+ `Identifiable`, `Insertable`, `AsChangeset` and `Associated`) are now
+ provided by `diesel_derives`. Furthermore those annotations are reexported
+ by `diesel` itself, so there is no need to depend on a second crate for
+ custom derives anymore. The `infer_schema!` and `infer_table_from_schema!`
+ macro is moved from diesel into the `diesel_infer_schema` crate. All
+ functionality related to migrations (The `diesel::migrations` module and the
+ `embed_migrations!` macro) is moved to the `diesel_migrations` crate. If you
+ are using any of those, simply depend on the new crates and your code will
+ continue to build.
```

Comment:
This is quite detailed and well written. I fear people will not read it, though… Can you add a small example of the change needed here?

> This means, if you were using `diesel_codegen` before, you can now replace this dependency
> with `diesel_infer_schema` (or remove it completely if you're not using `infer_schema!` in your code):
> 
> ```diff
> [dependencies]
> diesel = { version = "0.10.1", features = ["sqlite"] }
> -diesel_codegen = { version = "0.16", features = ["sqlite"] }
> +diesel_infer_schema = { version = "0.16", features = ["sqlite"] }
> ``` 
>
> and change the imports to
>
> ```rust
> #[macro_use] extern crate diesel;
> #[macro_use] extern crate diesel_infer_schema;
> ```
>
> (again, the second line is only necessary of you're using `infer_schema!`).
> In general, a lot of project will now only need to add diesel itself as a dependency.

Reply:
This is not that simple :wink: 
If someone is using `infer_schema!` he needs to import `diesel_infer_schema` now.
If someone is using `embed_migrations!` he needs to import `diesel_migrations` now.